### PR TITLE
Support release volcano in Travis

### PR DESCRIPTION
**Pull request by TommyLike**
This patch enables release volcano  when adding new tags. Things included.
1. Generate single yaml file which can be easily deployed within one command, see `helm template`.
2. Upload the docker images tagged with github tag.
3. Upload github release zip file.

Command for trigger a release would be something like:
```
git tag v0.0.2 -m/-f
git push origin v0.0.2
```
Tested on local repo:
image

And the structure of release file would be:
```
├── bin
│   └── amd64
│       ├── vc-admission
│       ├── vc-controllers
│       ├── vcctl
│       └── vc-scheduler
├── default-queue.yaml
├── helm
│   └── chart
│       └── volcano
│           ├── Chart.yaml
│           ├── config
│           │   ├── kube-batch-ci.conf
│           │   └── kube-batch.conf
│           ├── templates
│           │   ├── admission.yaml
│           │   ├── batch_v1alpha1_job.yaml
│           │   ├── bus_v1alpha1_command.yaml
│           │   ├── controllers.yaml
│           │   ├── default-queue.yaml
│           │   ├── NOTES.txt
│           │   ├── scheduler.yaml
│           │   ├── scheduling_v1alpha1_podgroup.yaml
│           │   ├── scheduling_v1alpha1_queue.yaml
│           │   ├── scheduling_v1alpha2_podgroup.yaml
│           │   └── scheduling_v1alpha2_queue.yaml
│           └── values.yaml
├── README.md
└── volcano-v0.0.6.yaml
```
Environment variables required in Travis:
1. DOCKER_PASSWORD
2. DOCKER_USERNAME
3. GITHUB_TOKEN

### CODE REVIEW

**Comment by TommyLike** [comment]
@k82cn  @hzxuzhonghu

**Comment by hzxuzhonghu** [comment]
good

**Comment by TommyLike** [comment]
@asifdxtreme  would you mind review and have a try on this patch? I see you published the first release.

**Comment by asifdxtreme** [comment]
> @asifdxtreme would you mind review and have a try on this patch? I see you published the first release.

okay I will have a check at that and updated it in sometime

File: hack/publish.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+set -o pipefail
+
+
+
+# The process of preparing volcano release.
+#   1. cp binaries into release folder
+#   2. cp README document into release folder
+#   3. cp default queue into release folder
+#   4. cp helm charts template into release folder and update default image tag
+#   5. upload docker images to volcano.sh
```
Comment (asifdxtreme) [inline hack/publish.sh]: Can we also add license file in the release?

**Comment by asifdxtreme** [comment]
@TommyLike Before actually releasing and uploading the images/release can we just use this release to deploy in kind and run E2E on this?

**Comment by k82cn** [comment]
do we need to have release in every build?

File: hack/publish.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+VK_ROOT=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
+BINARY_FOLDER=${VK_ROOT}/${BIN_DIR}/${REL_OSARCH}
+RELEASE_FOLDER=${VK_ROOT}/${RELEASE_DIR}
+RELEASE_BINARY=${RELEASE_FOLDER}/bin
+QUEUE_FILE=${VK_ROOT}/installer/helm/chart/volcano/templates/default-queue.yaml
+README_FILE=${VK_ROOT}/installer/README.md
+HELM_FOLDER=${VK_ROOT}/installer/helm
+VOLCANO_IMAGE_TAG=${TAG:-"latest"}
+DOCKER_PASSWORD=${DOCKER_PASSWORD:-""}
```
Comment (asifdxtreme) [inline hack/publish.sh]: we need to first add these environment variables in travis CI settings

**Comment by asifdxtreme** [comment]
> do we need to have release in every build?

Release will be triggered only if the new tag is pushed in github

File: hack/publish.sh (hunk 1 above)
Comment (TommyLike) [reply hack/publish.sh]: Great idea!

**Comment by TommyLike** [comment]
> @TommyLike Before actually releasing and uploading the images/release can we just use this release to deploy in kind and run E2E on this?

It's been verified before this steps. But only for helm chart, not the yaml way.

File: hack/publish.sh (hunk 2 above)
Comment (TommyLike) [reply hack/publish.sh]: Sure, it's mentioned above. I don't have the rights.

**Comment by asifdxtreme** [comment]
1. We need to skip the other stages which is not relevant for release in tagged commits
2. I tried this patch with my own fork, https://travis-ci.org/asifdxtreme/volcano/builds/559375980 , the docker images were pushed successfully, however the push binary to release failed.

**Comment by TommyLike** [comment]
> 1. We need to skip the other stages which is not relevant for release in tagged commits
> 2. I tried this patch with my own fork, https://travis-ci.org/asifdxtreme/volcano/builds/559375980 , the docker images were pushed successfully, however the push binary to release failed.

You need github token: GITHUB_TOKEN

**Comment by asifdxtreme** [comment]
> > 1. We need to skip the other stages which is not relevant for release in tagged commits
> > 2. I tried this patch with my own fork, https://travis-ci.org/asifdxtreme/volcano/builds/559375980 , the docker images were pushed successfully, however the push binary to release failed.
> 
> You need github token: GITHUB_TOKEN

my bad, ran it again after adding GITHUB_TOKEN and it works well https://travis-ci.org/asifdxtreme/volcano/builds/559381486
lgtm overall

File: hack/publish.sh (hunk 2 above)
Comment (asifdxtreme) [reply hack/publish.sh]: okay I have added it now
image

File: installer/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+## Prerequisites
+
+- Kubernetes 1.12+ with CRD support
+
+## Installing volcano via yaml file
+
+All-in-one yaml has been generated for quick deployment. Try command:
+```$xslt
+kubectl apply -f volcano-v0.0.x.yaml
```
Comment (hzxuzhonghu) [inline installer/README.md]: Add a guide on how to generate this yaml

Comment (TommyLike) [reply installer/README.md]: It should not be here, since it's only used for the releases and focus on how to use not how to develop.

Comment (hzxuzhonghu) [reply installer/README.md]: a step : make xxxx

Comment (TommyLike) [reply installer/README.md]: Since the source code is not downloaded in this scenario, it's better to add this steps into developing process.

**Pull request merged**